### PR TITLE
Assignment3: Jake Lucas & Ben Gales

### PR DESCRIPTION
**Pull request by JacobLucas02**
Assignment3: Jake Lucas and Ben Gales collaboration

@webnoben

### CODE REVIEW

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 1]
```diff
@@ -1,7 +1,89 @@
+using System.IO;
```
Comment (AustinH5544) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:1]: Consider making a couple tests without mock.

**Review by AustinH5544 — COMMENTED** [review]
Assignment
For this assignment do the following:

Create two interfaces. One to represent the JokeService and one interface to represent outputting the joke to the screen (Console.WriteLine). ✔
The JokeService will need to have the interface applied to it. ✔
Create an implementation of the output interface that writes out the joke to the console. ✔
Implement the Jester class. It should take in both interfaces as dependencies. These dependencies should be null checked. ✔
The Jester class TellJoke() method should retrieve a joke from the JokeService. If the joke contains "Chuck Norris", skip it and get another. The joke should be written to the output dependency. ✔
Unit test the Jester class. Code coverage should be above 90% for this class. Moq will make this significantly easier. ✔❌ **// You did unit test Jester but Code coverage could be higher.**
Fundamentals
Be sure you enable:
Nullability for all projects ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
Ensure that you turn on treat warnings as errors for all projects ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Ensure that you turn on EnforceCodeStyleInBuild ✔
For this assignment, always use Assert.Equal<T>() (the generic version) ✔
Ensure there are no errors or warnings (including code analysis warnings) ✔
All of the above should be unit tested ✔
Choose simplicity over complexity ✔
Extra Credit
Unit test your implementation that writes the joke out to the screen. How hard could it be to unit test a single line method ;)?❌✔ _**I do not believe you are doing this but I also do not know everything about mock**_
The Geek jokes API that is being used can also return jokes in a JSON format. Update the JokeService to retrieve jokes using JSON. GetJoke should still return a string.❌

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace CanHazFunny;
+
+public class Jester
+{
+    private IJokeService _jokeService;
+    private IOutputService _outputService;
+
```
[the remark this answers is not in the document]
Comment (JacobLucas02) [reply CanHazFunny/CanHazFunny/Jester.cs:9]: I forgot about that, thank you for the reminder!

File: CanHazFunny/CanHazFunny.Tests/IJokeServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using Moq;
+using Xunit;
+
+namespace CanHazFunny.Tests;
+
+public class IJokeServiceTests
```
Comment (AndrewKorchemniy) [inline CanHazFunny/CanHazFunny.Tests/IJokeServiceTests.cs:7]: We shouldn't test interfaces.

**Pull request closed without merging**